src/CsvTextEditor/ViewModels/MainViewModel.cs

```diff
@@ -35,34 +39,23 @@ public MainViewModel(IProjectManager projectManager)
         public Project Project { get; set; }
 
         #region Methods
-        protected override Task InitializeAsync()
+        protected override async Task InitializeAsync()
         {
-            _projectManager.ProjectActivatedAsync += OnProjectActivated;
+            await base.InitializeAsync();
 
-            return base.InitializeAsync();
+            _projectManager.ProjectActivationAsync += OnProjectActivationAsync;
         }
 
-        protected override Task CloseAsync()
+        private async Task OnProjectActivationAsync(object sender, ProjectUpdatingCancelEventArgs e)
         {
-            _projectManager.ProjectActivatedAsync -= OnProjectActivated;
-
-            return base.CloseAsync();
+            _dispatcherService.Invoke(() => Project = e.NewProject as Project, true);
```

Comment:
safe cast works a bit slower and if you don't need it, then don't use it.
in this project we always know that all projects have type `Project`, so no need to use safe type casting